### PR TITLE
TFW-2272 - implimented new notification bar styles

### PR DESCRIPTION
**Pull request by troyblank**
This does two things

- makes notification bars match the new styles in design (minus small screen breakpoints)
- updates the content to be less dependent on html by adding more specific props, making our reducer do most of the work and not having to do a lot of string concatenation on the fly.

I had to update some colors and icons during all this which is the majority of code changes you will see.

### CODE REVIEW

File: assets/sass/base/silents/_forms.scss [hunk 1]
```diff
@@ -98,7 +98,7 @@
         width: 0;
         height: 1px;
         transition: .2s ease width;
-        background-color: $color-blue;
+        background-color: $color-blue-1;
```
[the remark this answers is not in the document]
Comment (troyblank) [reply assets/sass/base/silents/_forms.scss]: I wanted to match what we are calling this color design wise, that way communication from design to development has no hicups. All our designs call this blue `blue 1`

[the remark this answers is not in the document]
Comment (tVeratti) [reply assets/sass/base/silents/_forms.scss]: Cool, makes sense to me.

File: assets/js/demo/components/notificationBar.js [hunk 2]
```diff
@@ -129,9 +150,12 @@ export default class NotificaitonBarDemo extends PureComponent {
             </button>
             <NotificationBar
               text={text}
```
[the remark this answers is not in the document]
Comment (troyblank) [reply assets/js/demo/components/notificationBar.js]: I don't think it hurts readability at all.

You are always so much more "one with the destructuring" than I! I like this, changed over to this format.

[the remark this answers is not in the document]
Comment (tVeratti) [reply assets/js/demo/components/notificationBar.js]: I am one with the destructuring, young skywalker.

**Pull request merged**